include/ofi_coll.h

```diff
@@ -72,73 +73,70 @@ enum coll_work_type {
 	UTIL_COLL_COMP,
 };
 
-struct util_coll_hdr {
-	struct slist_entry	entry;
-	struct slist_entry	barrier_entry;
-	enum coll_work_type	type;
-	/* only valid for xfer_item*/
-	uint64_t		tag;
-	int 			is_barrier;
+enum work_item_state {
+	WAITING,
+	PROCESSING
```

Comment:
add a prefix to the enum and states.  WAITING is too generic of a name that could result in conflicts.